### PR TITLE
Update Location to use whatwg-url's URLUtils

### PR DESCRIPTION
**Pull request by domenic**
A jsdom/env test is failing currently due to https://github.com/jsdom/whatwg-url/issues/31

### CODE REVIEW

**Comment by domenic** [comment]
whatwg-url-compat 0.6.1 now in use and fixes everything. Ready for review!

File: lib/jsdom/browser/location.js [hunk 1]
```diff
@@ -1,103 +1,67 @@
...
+Location.prototype.assign = function () {
+  notImplemented("location.assign", this[document]._defaultView);
+};
 
-  toString: function () {
-    return this._url.href;
-  },
+Location.prototype.replace = function (url) {
+  // This is nowhere near spec compliant, but has worked so far.
+  whatwgURL.setTheInput(this, url);
```
Comment (Sebmaster) [inline lib/jsdom/browser/location.js]: Wait wat?

File: lib/jsdom/browser/history.js [hunk 2]
```diff
@@ -11,10 +11,10 @@ function StateEntry(data, title, url) {
 module.exports = History;
 
 function History(window) {
-  this._states = [new StateEntry(null, "", window.location._url.href)];
+  this._states = [new StateEntry(null, "", window._document._URL)];
   this._index = 0;
   this._window = window;
-  this._location = window.location;
+  this._location = window._document._location;
```
Comment (Sebmaster) [inline lib/jsdom/browser/history.js]: Why go to document's location here?

File: lib/jsdom/browser/location.js (hunk 1 above)
Comment (domenic) [inline lib/jsdom/browser/location.js]: Not quite for replace. Replace generally does not cause a navigation. (It does, but with the replacement flag set.) The current code implements it as `this.href = val`.

File: lib/jsdom/browser/history.js (hunk 2 above)
Comment (domenic) [inline lib/jsdom/browser/history.js]: Because that's where it's really stored; window.location is a getter for this._document._location

**Comment by Sebmaster** [comment]
LGTM then.

**Pull request merged**